crates/pop-cli/tests/contract.rs

```diff
@@ -2,19 +2,57 @@
 
 use anyhow::Result;
 use assert_cmd::Command;
-use pop_common::{set_executable_permission, templates::Template};
+use pop_common::{find_free_port, set_executable_permission, templates::Template};
 use pop_contracts::{
 	contracts_node_generator, dry_run_gas_estimate_instantiate, instantiate_smart_contract,
 	run_contracts_node, set_up_deployment, Contract, UpOpts,
 };
-use std::{path::Path, process::Command as Cmd};
+use serde::{Deserialize, Serialize};
+use std::{path::Path, process::Command as Cmd, time::Duration};
 use strum::VariantArray;
+use subxt::{config::DefaultExtrinsicParamsBuilder as Params, tx::Payload, utils::to_hex};
+use subxt_signer::sr25519::dev;
+use tokio::time::sleep;
 use url::Url;
 
+// This struct implements the [`Payload`] trait and is used to submit
+// pre-encoded SCALE call data directly, without the dynamic construction of transactions.
+struct CallData(Vec<u8>);
+impl Payload for CallData {
+	fn encode_call_data_to(
+		&self,
+		_: &subxt::Metadata,
+		out: &mut Vec<u8>,
+	) -> Result<(), subxt::ext::subxt_core::Error> {
+		out.extend_from_slice(&self.0);
+		Ok(())
+	}
+}
+
+// TransactionData has been copied from wallet_integration.rs
+/// Transaction payload to be sent to frontend for signing.
+#[derive(Serialize, Debug)]
+#[cfg_attr(test, derive(Deserialize, Clone))]
+pub struct TransactionData {
```

Comment:
As a future refactor, we make this struct available from `pop-cli` and remove the need of duplication.